core/commands/pin.go

```diff
@@ -215,20 +233,62 @@ collected if needed. (By default, recursively. Use -r=false for direct pins.)
 			return err
 		}
 
-		removed, err := corerepo.Unpin(n, api, req.Context, req.Arguments, recursive)
-		if err != nil {
-			return err
+		for _, b := range req.Arguments {
+			p, err := coreiface.ParsePath(b)
+			if err != nil {
+				return err
+			}
+
+			rp, err := api.ResolvePath(req.Context, p)
+			if err != nil {
+				return err
+			}
+
+			if err := api.Pin().Rm(req.Context, rp, options.Pin.RmRecursive(recursive)); err != nil {
+				if err := res.Emit(&PinOutput{
```

Comment:
This is still going to break js-ipfs-api.

I'd keep a list of strings with `rp.Cid().String()` and use `Error map[string]error` for tracking errors, and at the end just do `cmds.EmitOnce(&PinOutput{pins, errors})`.

This is somewhat hacky, but should work. Maybe @Stebalien has some better idea